deployments/container/Dockerfile

```diff
@@ -12,47 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# We use an ubuntu20.04 base image to allow for a more efficient multi-arch builds.
-FROM  nvcr.io/nvidia/cuda:12.8.1-base-ubuntu20.04 AS build
+# Run build with binaries native to the current build platform.
+FROM --platform=$BUILDPLATFORM nvcr.io/nvidia/cuda:12.8.1-base-ubuntu20.04 AS build
 
-RUN apt-get update && \
-    apt-get install -y wget make git gcc-aarch64-linux-gnu gcc \
-    && \
-    rm -rf /var/lib/apt/lists/*
-
-# Require build arg.
+# Require GOLANG_VERSION arg to be provided.
 ARG GOLANG_VERSION=x.x.x
 
-RUN set -eux; \
-    \
-    arch="$(uname -m)"; \
-    case "${arch##*-}" in \
-        x86_64 | amd64) ARCH='amd64' ;; \
-        ppc64el | ppc64le) ARCH='ppc64le' ;; \
-        aarch64) ARCH='arm64' ;; \
-        *) echo "unsupported architecture" ; exit 1 ;; \
-    esac; \
-    wget -nv -O - https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-${ARCH}.tar.gz \
+# BUILDARCH, TARGETARCH, etc are defined in the global scope by BuiltKit.
+# BUILDARCH is reflecting the architecture of the build platform. TARGETARCH is
+# set via the --platform arg provided to the `docker buildx build ...` command.
+# Redefining those variables here without new values makes the outer-context
+# values available to in-stage RUN commands. Arch values are of the form
+# amd64/arm64.
+ARG BUILDARCH
+ARG TARGETARCH
+
+RUN apt-get update && \
+    apt-get install -y wget make git gcc-aarch64-linux-gnu gcc && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
I know that this was just moved, but does it make sense to have one dependency on a line to make future diffs cleaner?

Reply:
done